src/app/campaign/store/reducers/campaign.reducer.ts

```diff
@@ -26,6 +26,26 @@ export function reducer(state = initialState, action: CampaignActions | Advertis
     case ActionTypes.CAMPAIGN_NEW: {
       return { ...initialState, loading: false, loaded: true };
     }
+    case ActionTypes.CAMPAIGN_DUP_FROM_FORM: {
+      return {
+        localCampaign: duplicateCampaign(action.payload.campaign),
+        changed: true,
+        valid: true,
```

Comment:
These are valid?  I guess if you were only allowing already-valid campaigns to be duplicated.

Reply:
It's only valid campaigns that can be duplicated. @brandonhundt and I talked this one out https://prx.slack.com/archives/GCEJE5136/p1586446627005200?thread_ts=1586444042.004000&cid=GCEJE5136 

Also I dont have any validations written to validate these models except what we get from the reactive form, so I have to assume either valid or invalid or duplicate all the reactive form validations. Maybe we will need to have separate validations against the model at some point, but I dont think this warrants it just yet.

Reply:
+1 makes sense.